### PR TITLE
Use `channel` instead of the removed `type` for axis's rules

### PR DESCRIPTION
**Pull request by kanitw**

### CODE REVIEW

**Comment by kanitw** [comment]
I changed channel === X to also consider COLUMN.  This should be mergable once the test passes.  

Note that I also rename commit message to start with a verb to conform with our PR guide :) 
https://github.com/vega/vega-lite/blob/master/.github/PULL_REQUEST_TEMPLATE.md

File: test/compile/axis.test.ts [hunk 1]
```diff
@@ -503,7 +503,7 @@ describe('Axis', function() {
           x: {field: "a", type: "temporal", timeUnit: "month"}
         }
       });
-      const labels = axis.encode.labels(model, X, {}, {type: 'x'});
+      const labels = axis.encode.labels(model, X, {}, {});
```
Comment (domoritz) [inline test/compile/axis.test.ts]: Wanna test COLUMN as well?

Comment (kanitw) [inline test/compile/axis.test.ts]: You can help add that :p

Comment (domoritz) [inline test/compile/axis.test.ts]: I did

**Comment by kanitw** [comment]
FYI, test is failing ..

**Comment by domoritz** [comment]
Jep. I know.

**Comment by kanitw** [comment]
LGTM -- not sure if you wanna merge it yet.

**Pull request merged**